scripts/src/chartprreview/chartprreview.py

```diff
@@ -221,6 +221,15 @@ def check_report_success(directory, report_path, version):
         write_error_log(directory, msg)
         sys.exit(1)
 
+    if "charts.openshift.io/certifiedOpenShiftVersions" in annotations:
+        pattern = re.compile("(\d+).(\d+).*")
```

Comment:
not sure if the reg is correct - `.` matches anything not only dots.

there are plenty of libraries that can be used for parsing versions - for example: https://pypi.org/project/semver/ -

Reply:
The input is coming from the chart-verifier tool.

1. It could be either an output of [oc version command](https://github.com/redhat-certification/chart-verifier/blob/511119c1af7081392bf40f440debc20948268983/pkg/tool/oc.go#L23)
2. Alternative could be a direct [input from the user which is validated for SemVer spec](https://github.com/redhat-certification/chart-verifier/blob/511119c1af7081392bf40f440debc20948268983/pkg/chartverifier/certifier.go#L88).

In this case, we are interested in the `MAJOR.MINOR` part of the version to set as an annotation.  So, I don't see a need to verify the version for SemVer spec again.

Reply:
We want to check if the value is according to semver format, and the current code is using regex pattern that matches strings that are not semver versions, take a look at this:

```python
mport re

v = "1a2"

pattern = re.compile("(\d+).(\d+).*")
match = pattern.match(v)
if match:
    print("matched")
```

Let's not reinvent the wheel here - instead we can use some library for parsing the version and implicitly validating it. https://pypi.org/project/semver/ is just one of them

Reply:
Done. 89b9639d6d5a12a7c6985ee59479594fabf69a31